### PR TITLE
Parse high-res icons fixes #85

### LINKED ISSUE #85: Find the largest high res icon
When we parse for high res icons, like apple touch icons, we need to take into account the size properties and sort for the largest one.

### PR DESCRIPTION
**Pull request by jaredkerim**

### CODE REVIEW

**Comment by rlr** [comment]
This looks good. I think we might want a special case for when `sizes="any"`. For example, I say this on yahoo: `<link rel="icon" href="https://s.yimg.com/os/mit/media/p/common/images/favicon_new-7483e38.svg" sizes="any">`. Although, now that I think about it, if they have an svg like this they are very unlikely to have more than one size listed. I think I've convinced myself that you maybe don't need to handle that.

File: parser.js [hunk 1]
```diff
@@ -73,6 +87,15 @@ const metadataRules = {
       ['link[rel="Shortcut Icon"]', node => node.element.getAttribute('href')],
       ['link[rel="mask-icon"]', node => node.element.getAttribute('href')],
     ],
+    scorers: [
+      (node, score) => {
+        const sizes = node.element.getAttribute('sizes');
+
+        if (sizes) {
+          return score + sizes.match(/\d+/g).reduce((a, b) => a * b);
```
Comment (rlr) [inline parser.js]: Oh, but I tried this in the console:

```
>> "any".match(/\d+/g).reduce((a,b) => a*b);
x TypeError: "any".match(...) is null 
```

So you should add some handling `size` attribute values that aren't "NxM"

File: parser.js [hunk 2]
```diff
@@ -73,6 +87,15 @@ const metadataRules = {
...
+        }
+      }
```
Comment (pdehaan) [inline parser.js]: Do we need an `else` here or anything? (ie: if `sizes` is falsey, do we return original `score` or whatever?)

**Comment by rlr** [comment]
💃

**Pull request merged**